lib/kitchen/provisioner/chef_base.rb

```diff
@@ -115,6 +131,28 @@ class ChefBase < Base
       def initialize(config = {})
         super(config)
 
+        if config.key?(:require_chef_omnibus)
+          add_config_deprecation! :warn, "require_chef_omnibus", <<-EOF
+  To install a specific version use 'product_version' along with the 'product_name' and 'channel' settings.
+  To skip the provisioner bootstrap installation set 'skip_bootstrap' to true.
+EOF
+        elsif config[:skip_bootstrap] == true
+          # New setting that will replace multi-use require_chef_omnibus for skipping bootstrap installations.
+          config[:require_chef_omnibus] = false
+        else
+          # Ensure this is set to true if not set. product_name will take precedence when set.
+          config[:require_chef_omnibus] = true
+        end
+
+        if config.key?(:chef_omnibus_url)
+          add_config_deprecation! :warn, "chef_omnibus_url", <<-EOF
+  To override the install script url use 'install_script_url'.
+  Note that Windows will now automatically use the appropriate install.ps1 script.
+EOF
+        else
+          config[:chef_omnibus_url] = config[:install_script_url]
```

Comment:
Looks like this broke script_generator.
https://github.com/chef/mixlib-install/blob/master/lib/mixlib/install/script_generator.rb#L213-L218
I'll look into this, but we may have to set the url to install.sh like before as default.